### DISCUSSION TITLE
Examples for creating Resources and Artifacts other than REST Resources

### DISCUSSION THREAD

**Discussion by sebplorenz** [International-Data-Spaces-Association/DataspaceConnector#63] [Q&A] [answered]
Hi, the current example at Hands-on IDS Communication shows, how a resource/artifact can be created using a REST Service. Are there other examples for artifact creation? Like serving a local file as an artifact or storing data into the DSC DB and serve the data as a resource/artifact?

**Comment by juliapampus** [comment, marked as answer]
Yes, you can set the representation's source type to `LOCAL` and use the endpoint `POST /admin/api/resources/{resource-id}/data` to push a string that will be saved directly to the connector's database. Unfortunately, you cannot upload a file directly, you would have to encode it first - open todo on our list.
